### PR TITLE
[Beta] Consume @azure/monitor-opentelemetry

### PR DESCRIPTION
**Pull request by hectorhdzg**
Removing all unnecessary code from here
This project would keep following functionalities:
Auto log collection
Native Metrics
Agents code
Shim

### CODE REVIEW

File: src/applicationInsightsConfig.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { InstrumentationOptions } from "@azure/monitor-opentelemetry";
+
+/** 
+* @deprecated Use ApplicationInsightsOptions instead
+*/
+export class ApplicationInsightsConfig implements ApplicationInsightsOptions {
+    // ApplicationInsightsOptions
+    public otlpTraceExporterConfig?: OTLPExporterConfig;
+    public otlpMetricExporterConfig?: OTLPExporterConfig;
+    public  enableAutoCollectExceptions?: boolean;
```
Comment (JacksonWeber) [inline src/applicationInsightsConfig.ts:17]:
```suggestion
    public enableAutoCollectExceptions?: boolean;
```
Double spacing here

File: src/types.ts [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+   */
+    enableAutoCollectExceptions?: boolean;
+    /**
+    * Log Instrumentations configuration included as part of Application Insights (console, bunyan, winston)
+    */
+    logInstrumentations?: LogInstrumentationsConfig;
+     /**
+     * Specific extended metrics, applicationinsights-native-metrics package need to be available
+     */
+     extendedMetrics?: { [type: string]: boolean };
```
Comment (JacksonWeber) [inline src/types.ts:32]:
```suggestion
    logInstrumentations?: LogInstrumentationsConfig;
    /**
    * Specific extended metrics, applicationinsights-native-metrics package need to be available
    */
    extendedMetrics?: { [type: string]: boolean };
```

File: package.json [hunk 3]
```diff
@@ -69,6 +69,7 @@
     "@azure/core-client": "^1.0.0",
     "@azure/core-rest-pipeline": "^1.9.2",
     "@azure/identity": "^3.1.3",
+    "@azure/monitor-opentelemetry": "file:azure-monitor-opentelemetry-0.0.0-beta.0.tgz",
```
Comment (JacksonWeber) [inline package.json:72]: Was this just being used for testing? If so can we remove it?

File: src/shim/configuration/jsonConfig.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        // JSON string added directly in env variable
+        if (contentJsonConfig) {
+            jsonString = contentJsonConfig;
+        }
+        // JSON file
+        else {
+            let configFileName = "applicationinsights.json";
+            let rootPath = path.join(__dirname, "../../../"); // Root of folder (__dirname = ../dist-esm/src)
+            let tempDir = path.join(rootPath, configFileName); // default
+            let configFile = process.env[ENV_CONFIGURATION_FILE];
```
Comment (JacksonWeber) [inline src/shim/configuration/jsonConfig.ts:40]:
```suggestion
            const configFileName = "applicationinsights.json";
            const rootPath = path.join(__dirname, "../../../"); // Root of folder (__dirname = ../dist-esm/src)
            const tempDir = path.join(rootPath, configFileName); // default
            const configFile = process.env[ENV_CONFIGURATION_FILE];
```

File: src/types.ts [hunk 5]
```diff
@@ -1,21 +1,45 @@
...
+ */
+export interface ApplicationInsightsOptions extends AzureMonitorOpenTelemetryOptions {
+    /** OTLP Trace Exporter Configuration */
+    otlpTraceExporterConfig?: OTLPExporterConfig;
+    /** OTLP Metric Exporter Configuration */
+    otlpMetricExporterConfig?: OTLPExporterConfig;
+    /**
+   * Sets the state of exception tracking (enabled by default)
+   * if true uncaught exceptions will be sent to Application Insights
+   */
```
Comment (JacksonWeber) [inline src/types.ts:23]:
```suggestion
    /**
    * Sets the state of exception tracking (enabled by default)
    * if true uncaught exceptions will be sent to Application Insights
    */
```

**Review by JacksonWeber — APPROVED** [review]
LGTM

File: package.json (hunk 3 above)
Comment (hectorhdzg) [reply package.json:72]: Package doesn't exist in npm, that is why this PR is still a draft, will update when available

**Pull request merged**